src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,59 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  takeDamage(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((item) => item !== this);
+
+    return `${this.name}`;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  // eslint-disable-next-line no-useless-constructor
+  constructor(name, health) {
+    super(name, health);
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
```

Comment:
[FUNCTION NAMING] - The `hide` method should set the `hidden` property to `true` but not toggle its value. The current implementation toggles the value, which is not as per the specifications.